src/motion_input_shaping/shaping_demo.py

```diff
@@ -77,11 +81,14 @@ def demo_shaping_class(
         try:
             num_lockstep_groups_possible = device.settings.get("lockstep.numgroups")
             for group_num in range(1, int(num_lockstep_groups_possible) + 1):
-                axis_nums = device.get_lockstep(group_num).get_axis_numbers()
-                if AXIS_INDEX in axis_nums:
-                    print(f"Axis {AXIS_INDEX} is part of Lockstep group {group_num}.")
-                    LOCKSTEP_INDEX = group_num
-                    break
+                try:
+                    axis_nums = device.get_lockstep(group_num).get_axis_numbers()
+                    if AXIS_INDEX in axis_nums:
+                        print(f"Axis {AXIS_INDEX} is part of Lockstep group {group_num}.")
+                        LOCKSTEP_INDEX = group_num
+                        break
+                except LockstepNotEnabledException:
+                    pass
```

Comment:
You can simplify the try catch block by catching more than one exception at the same level (reduce nestedness)
https://stackoverflow.com/questions/6470428/catch-multiple-exceptions-in-one-line-except-block

Reply:
I am not going to make this change yet because I don't know all the logic behind when this exception is thrown and I think it is safer having the separate try catch block within the for loop. At the moment, if the LockstepNotEnabledException occurs when checking lockstep group 1 it will move on to check the remaining lockstep groups.

Reply:
Ahh good point!  I missed that there is a for loop in there and it is checking more than one lockstep group.